elementary/monitor/api/models/models.py

```diff
@@ -1,16 +1,86 @@
+from collections import defaultdict
 import json
 import os
-from typing import Dict, Optional, Union
+import statistics
+from typing import Dict, List, Optional, Union
 
 from elementary.clients.api.api import APIClient
-from elementary.monitor.api.models.schema import ExposureSchema, ModelCoverageSchema, ModelSchema, NormalizedExposureSchema, NormalizedModelSchema, NormalizedSourceSchema, SourceSchema
+from elementary.monitor.api.models.schema import (
+    ExposureSchema,
+    ModelCoverageSchema,
+    ModelRunSchema,
+    ModelRunsSchema,
+    ModelSchema,
+    NormalizedExposureSchema,
+    NormalizedModelSchema,
+    NormalizedSourceSchema,
+    SourceSchema,
+    TotalsModelRunsSchema
+)
 from elementary.utils.json_utils import try_load_json
+from elementary.utils.log import get_logger
+
+logger = get_logger(__name__)
 
 YAML_FILE_EXTENSION = ".yml"
 SQL_FILE_EXTENSION = ".sql"
 
 
 class ModelsAPI(APIClient):
+    def get_models_runs(self, days_back: Optional[int] = 7) -> List[ModelRunsSchema]:
+        run_operation_response = self.dbt_runner.run_operation(macro_name='get_models_runs',
+                                                               macro_args=dict(days_back=days_back))
+        model_run_dicts = json.loads(run_operation_response[0]) if run_operation_response else []
+        
+        models_runs = defaultdict(list)
+        for model_run in model_run_dicts:
+            models_runs[model_run['unique_id']].append(model_run)
+        
+        aggregated_models_runs = []
+        for model_unique_id, model_runs in models_runs.items():
+            totals = self._get_model_runs_totals(model_runs)
+            runs = [
+                ModelRunSchema(
+                    id=model_run['invocation_id'],
+                    time_utc=model_run['generated_at'],
+                    status=model_run['status'],
+                    full_refresh=model_run['full_refresh'],
+                    materialization=model_run['materialization'],
+                    execution_time=model_run['execution_time']
+                )
+                for model_run
+                in model_runs
+            ]
+            # We want to keep failed model runs out from the median.
+            execution_times_for_median = [model_run['execution_time'] for model_run in model_runs if model_run['status'] != 'error']
```

Comment:
I think we should exclude skipped models as well so I think a better filtering could be - 
`if lower(model_runs['status']) == 'success'`